src/ts/components/touchcontroloverlay.ts

```diff
@@ -0,0 +1,175 @@
+    });
+
+    this.config = this.mergeConfig(config, {
+      cssClass: 'ui-touchcontrol-overlay',
+      acceptsTouchWithUiHidden: true,
+      seekTime: 10,
+      seekDoubleTapMargin: 15,
+      components: [this.playbackToggleButton],
+    }, this.config);
+  }
+
+  configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+    super.configure(player, uimanager);
+
+    this.doubleTapTimeout = new Timeout(500, () => {
+      this.couldBeDoubleTapping = false;
+    });
+
+    uimanager.onControlsHide.subscribe(() => {
+      this.playbackToggleButton.hide();
+    });
+
+    uimanager.onControlsShow.subscribe(() => {
+      this.playbackToggleButton.show();
+    });
+
+    let playerSeekTime = 0;
+
+    this.touchControlEvents.onSeekBackward.subscribe(() => {
+      playerSeekTime -= this.config.seekTime;
+      player.seek(playerSeekTime);
+    });
+
+    this.touchControlEvents.onSeekForward.subscribe(() => {
+      playerSeekTime += this.config.seekTime;
+      player.seek(playerSeekTime);
+    });
+
+    this.touchControlEvents.onSingleClick.subscribe((_, e) => {
+      uimanager.getUI().toggleUiShown();
+      playerSeekTime = player.getCurrentTime();
+
+      const eventTarget = (e as Event).target as HTMLElementWithComponent;
+      const rect = eventTarget.getBoundingClientRect();
+      const eventTapX = ((<any>e).clientX) - rect.left;
+      const eventTapY = ((<any>e).clientY) - rect.top;
+      this.latestTapPosition = {x: eventTapX, y: eventTapY};
+    });
+
+    this.touchControlEvents.onDoubleClick.subscribe((_, e) => {
+      uimanager.getUI().hideUi();
+      const event = e as Event;
+      const eventTarget = event.target as HTMLElementWithComponent;
+      if (!eventTarget || !((eventTarget).component instanceof TouchControlOverlay)) {
+        return;
+      }
+
+      const width = eventTarget.clientWidth;
+      const tapMargin = width * 0.4;
+      const rect = eventTarget.getBoundingClientRect();
+      const eventTapX = ((<any>e).clientX) - rect.left;
+      const eventTapY = ((<any>e).clientY) - rect.top;
+      if (this.latestTapPosition.x === -1 && this.latestTapPosition.y === -1) {
+        this.latestTapPosition = {x: eventTapX, y: eventTapY};
+      }
+      const doubleTapMargin = this.config.seekDoubleTapMargin;
+      if (Math.abs(this.latestTapPosition.x - eventTapX) <= doubleTapMargin && Math.abs(this.latestTapPosition.y - eventTapY) <= doubleTapMargin)
+        if (eventTapX < tapMargin) {
+          this.touchControlEvents.onSeekBackward.dispatch(this);
+        }
+        else if (eventTapX > (width - tapMargin)) {
+          this.touchControlEvents.onSeekForward.dispatch(this);
+        }
+      this.latestTapPosition = {x: eventTapX, y: eventTapY};
+    });
+
+    const thisDomElement = this.getDomElement();
+    thisDomElement.on('click', (e) => {
+      if ((e.target as HTMLElementWithComponent).component instanceof TouchControlOverlay) {
```

Comment:
you already did `const eventTarget = event.target as HTMLElementWithComponent` before soe
```suggestion
      if (eventTarget.component instanceof TouchControlOverlay) {
```
could get rid of the cast.

Reply:
yeah, but not in the scope of the click listener

Reply:
Oooohhh right. that is a different one, my bad.